smoke-tests/src/test/java/io/lumigo/javaagent/smoketest/SpringBootSmokeTest.java

```diff
@@ -29,6 +33,20 @@
 import org.junit.jupiter.api.Test;
 
 class SpringBootSmokeTest extends SmokeTest {
+  @Test
+  public void testResourceProviders() throws IOException {
+    URL url =
+        new URL(
+            "jar:file:"
+                + agentPath
+                + "!/inst/META-INF/services/io.opentelemetry.sdk.autoconfigure.spi.ResourceProvider");
+    InputStream is = url.openStream();
```

Comment:
The lack of try-with-resource usage is going to make the code analyzer weep, please fix it ;-)

Reply:
The whole method can return an IOException (which will fail the test), we don't need try-catch...